intermittent test failure, `'~' contains invalid characters.`

e.g. https://travis-ci.org/gratipay/gratipay.com/builds/76386460

```
=================================== FAILURES ===================================
_______________ Tests.test_changing_username_to_restricted_name ________________
self = <test_participant.Tests testMethod=test_changing_username_to_restricted_name>
    def test_changing_username_to_restricted_name(self):
        with self.assertRaises(UsernameIsRestricted):
>           self.participant.change_username(self.random_restricted_username())
tests/py/test_participant.py:409: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <Participant u'user1'>, suggested = '~'
    def change_username(self, suggested):
        """Raise Response or return None.
    
            Usernames are limited to alphanumeric characters, plus ".,-_:@ ",
            and can only be 32 characters long.
    
            """
        # TODO: reconsider allowing unicode usernames
        suggested = suggested and suggested.strip()
    
        if not suggested:
            raise UsernameIsEmpty(suggested)
    
        if len(suggested) > USERNAME_MAX_SIZE:
            raise UsernameTooLong(suggested)
    
        if set(suggested) - ASCII_ALLOWED_IN_USERNAME:
>           raise UsernameContainsInvalidCharacters(suggested)
E           UsernameContainsInvalidCharacters: The username '~' contains invalid characters.
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25969625-intermittent-test-failure-contains-invalid-characters?utm_campaign=plugin&utm_content=tracker%2F85909&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F85909&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/tests/py/test_participant.py
+++ b/tests/py/test_participant.py
@@ -348,7 +348,7 @@ class Tests(Harness):
         """helper method to chooses a restricted username for testing """
         from gratipay import RESTRICTED_USERNAMES
         random_item = random.choice(RESTRICTED_USERNAMES)
-        while random_item.startswith('%'):
+        while any(map(random_item.startswith, ('%', '~'))):
             random_item = random.choice(RESTRICTED_USERNAMES)
         return random_item
 

